### PR TITLE
Add --optimize option to ctr-remote image convert

### PR DESCRIPTION
**Pull request by ktock**
#230
#223

This commit adds `--optimize` option to ctr-remote image convert for enabling to
optimize the input image. This adds new package `analyzer` for analyzing layers
and dertermining prioritized files. Currently, FUSE-based analyzer is only
implemented but we can have other implementations (e.g. FUSEless analyzer) in
the future.

The following changes will come after this PR:
- having more analyzer e.g. FUSEless analyzer, static analyzer, containerd-based analyzer, etc.
- having converter command runnable without containerd daemon
  - modifying `ctr-remote image optimize` based on native converter + containerd's registry lib?

### CODE REVIEW

File: cmd/ctr-remote/commands/convert.go [hunk 1]
```diff
@@ -66,6 +75,14 @@ When '--all-platforms' is given all images in a manifest list must be available.
 			Usage: "eStargz chunk size",
 			Value: 0,
 		},
+		cli.BoolFlag{
+			Name:  "reuse",
```
Comment (AkihiroSuda) [inline cmd/ctr-remote/commands/convert.go:79]: estargz flags should have `estargz-` prefix as in existing flags.

File: cmd/ctr-remote/commands/convert.go [hunk 2]
```diff
@@ -85,19 +102,31 @@ When '--all-platforms' is given all images in a manifest list must be available.
 			Name:  "all-platforms",
 			Usage: "exports content from all platforms",
 		},
-	},
-	Action: func(context *cli.Context) error {
+	}, samplerFlags...),
```
Comment (AkihiroSuda) [inline cmd/ctr-remote/commands/convert.go:105]: I don't like this design.
`ctr images convert` should remain pure static image converter.
`ctr optimize` can have exec flags as in the current revision, but I think it should be reimplemented with `ctr image mount + ctr run + ctr images convert`.

Comment (ktock) [reply cmd/ctr-remote/commands/convert.go:105]: Sure. I'll reconsider the design of `ctr optimize` to make it containerd-based.

**Comment by ktock** [comment]
Closing in favor of #240.

**Pull request closed without merging**